zerver/models.py

```diff
@@ -2068,7 +2069,9 @@ class AbstractMessage(models.Model):
 
     # A JSON-encoded list of objects describing any past edits to this
     # message, oldest first.
-    edit_history: Optional[str] = models.TextField(null=True)
+    edit_history: Optional[List[Dict[str, Any]]] = ArrayField(
+        models.JSONField(null=True), null=True
```

Comment:
I don't think we want the inner JSONField to be `null=True` -- we just want to support empty arrays, not arrays containing empty JSON objects.

Reply:
I think also the last commit should remove the `null=True` on the array and replace it with `default=list` (i.e. an empty list)?  Not sure if that's a good idea.

Reply:
In the documentation it is given that:
> using default=[] creates a mutable default that is shared between all instances of ArrayField.

that is why, i kept it as a null and also keeping it null helped in keeping the code more or less same, I just had to get rid of the `orjson.dumps` and `orjson.loads` and it worked just fine.

Reply:
Yes, but `default=list` is recommended, while `default=[]` is not.